test/base/module/record/Record.addAllRecords.test.js

```diff
@@ -0,0 +1,235 @@
+      expectResults.push({'results': []});
+      for (let index = 0; index < loopTimes; index++) {
+        const start = index * UPDATE_RECORDS_LIMIT;
+        const end = NUMBER_RECORDS - (start + UPDATE_RECORDS_LIMIT) < UPDATE_RECORDS_LIMIT ? NUMBER_RECORDS : start + UPDATE_RECORDS_LIMIT;
+        if (i < (numBulk - 1) && expectBodys[i].requests.length < limitRequest || i === (numBulk - 1) && (index < (loopTimes - (i * limitRequest)))) {
+          expectBodys[i].requests.push({
+            'api': API_ROUTE.RECORDS,
+            'method': 'POST',
+            'payload': {
+              'app': appID,
+              'records': recordsData.slice(start, end)
+            }
+          });
+          expectResults[i].results.push({
+            'ids': resultsIds.slice(start, end),
+            'revisions': resultsRevisons.slice(start, end)
+          });
+        }
+      }
+      finalResults.results = finalResults.results.concat(expectResults[i].results);
+    }
+
+    nock(URI)
+      .post(API_ROUTE.BULK, (rqBody) => {
+        expect(rqBody).toHaveProperty('requests');
+        expect(rqBody).toEqual(expectBodys[0]);
+        return true;
+      })
+      .matchHeader('Content-Type', (type) => {
+        expect(type).toEqual(expect.stringContaining('application/json'));
+        return true;
+      })
+      .reply(200, expectResults[0])
+      .post(API_ROUTE.BULK, (rqBody) => {
+        expect(rqBody).toHaveProperty('requests');
+        expect(rqBody).toEqual(expectBodys[1]);
+
+        return true;
+      })
+      .matchHeader('Content-Type', (type) => {
+        expect(type).toEqual(expect.stringContaining('application/json'));
+        return true;
+      })
+      .reply(200, expectResults[1]);
+    const addRecordsResult = recordModule.addAllRecords({app: appID, records: recordsData});
+    return addRecordsResult.then((rsp) => {
+      expect(rsp).toHaveProperty('results');
+      expect(rsp.results).toMatchObject(finalResults.results);
+      expect(rsp.results[0].ids.length).toEqual(finalResults.results[0].ids.length);
+      expect(rsp.results[0].revisions.length).toEqual(finalResults.results[0].revisions.length);
+    });
+  });
+
+  it('should throw error when fail to add records', () => {
+    const appID = 99999;
+    const recordsData = [];
+    recordsData.push({
+      Text: {
+        value: 'test'
+      }
+    });
+    const expectResult = {
+      'results': [
+        {
+          'message': 'Record(id:appID)NotFound.',
+          'id': '1505999166-1721668264',
+          'code': 'GAIA_RE01'
+        }
+      ]
+    };
+    nock(URI)
+      .post(API_ROUTE.BULK, (rqBody) => {
+        expect(rqBody).toHaveProperty('requests');
+        return true;
+      })
+      .reply(400, expectResult);
+
+    const addRecordResult = recordModule.addAllRecords({app: appID, records: recordsData});
+    return addRecordResult.catch((err) => {
+      expect(err.results[0]).toBeInstanceOf(KintoneAPIException);
```

Comment:
@phongnhat19 
Please, refactor this failed case. expectResult should be renamed to response.
And should also define expectedErrorResponse to match with err.results[0].errorResponse
for example:
it('should throw error when fail to add records', () => {
    const appID = 99999;
    const recordsData = [];
    recordsData.push({
      Text: {
        value: 'test'
      }
    });
    const response = {
      'results': [
        {
          'message': 'Record(id:appID)NotFound.',
          'id': '1505999166-1721668264',
          'code': 'GAIA_RE01'
        }
      ]
    };
    const expectedError = {
      'message': 'Record(id:appID)NotFound.',
      'id': '1505999166-1721668264',
      'code': 'GAIA_RE01'
    };
    nock(URI)
      .post(API_ROUTE.BULK, (rqBody) => {
        expect(rqBody).toHaveProperty('requests');
        return true;
      })
      .reply(400, response);

    const addRecordResult = recordModule.addAllRecords({app: appID, records: recordsData});
    return addRecordResult.catch((err) => {
      expect(err.results[0].errorResponse).toMatchObject(expectedError);
    });